test/built-ins/Object/defineProperty/15.2.3.6-4-591.js

```diff
@@ -9,51 +9,61 @@ es5id: 15.2.3.6-4-591
 description: >
     ES5 Attributes - Fail to update value of property of
     [[Proptotype]] internal property (Object.create)
-includes: [runTestCase.js]
+includes: [propertyHelper.js]
 ---*/
 
-function testcase() {
-        var appointment = {};
-
-        var data1 = 1001;
-        Object.defineProperty(appointment, "startTime", {
-            get: function () {
-                return data1;
-            },
-            enumerable: false,
-            configurable: false
-        });
-        var data2 = "NAME";
-        Object.defineProperty(appointment, "name", {
-            get: function () {
-                return data2;
-            },
-            enumerable: false,
-            configurable: true
-        });
-
-        var meeting = Object.create(appointment);
-        var data3 = "In-person meeting";
-        Object.defineProperty(meeting, "conferenceCall", {
-            get: function () {
-                return data3;
-            },
-            enumerable: false,
-            configurable: false
-        });
-
-        var teamMeeting = Object.create(meeting);
-        teamMeeting.name = "IE Team Meeting";
-        var dateObj = new Date("10/31/2010 08:00");
-        teamMeeting.startTime = dateObj;
-        teamMeeting.conferenceCall = "4255551212";
-
-        var hasOwnProperty = !teamMeeting.hasOwnProperty("name") &&
-            !teamMeeting.hasOwnProperty("startTime") &&
-            !teamMeeting.hasOwnProperty('conferenceCall');
-
-        return hasOwnProperty && teamMeeting.name === "NAME" &&
-            teamMeeting.startTime === 1001 &&
-            teamMeeting.conferenceCall === "In-person meeting";
-    }
-runTestCase(testcase);
+var appointment = {};
+
+var data1 = 1001;
+Object.defineProperty(appointment, "startTime", {
+    get: function () {
+        return data1;
+    },
+    enumerable: false,
+    configurable: false
+});
+var data2 = "NAME";
+Object.defineProperty(appointment, "name", {
+    get: function () {
+        return data2;
+    },
+    enumerable: false,
+    configurable: true
+});
+
+var meeting = Object.create(appointment);
+var data3 = "In-person meeting";
+Object.defineProperty(meeting, "conferenceCall", {
+    get: function () {
+        return data3;
+    },
+    enumerable: false,
+    configurable: false
+});
+
+var teamMeeting = Object.create(meeting);
+
+verifyNotWritable(teamMeeting, "name", "nocheck");
+verifyNotWritable(teamMeeting, "startTime", "nocheck");
+verifyNotWritable(teamMeeting, "conferenceCall", "nocheck");
+
+try {
+    teamMeeting.name = "IE Team Meeting";
+} catch (e) {}
```

Comment:
I was a bit confused by this one, but I think it's a test that grew from a real user experience  (hence the unusually real-looking data).  

I added verifyNotWritable() because it is strict/nonstrict safe.  But I also chose to preserve the assignments (eg

```
teamMeeting.name = "IE Team Meeting";
```

I then had to wrap each of those in a `try` for strict mode.  I could add an assertion to the catch block to make it nonempty...